### ISSUE TITLE
sampler.lnprobability is transposed in emcee3?

### ISSUE THREAD

**Issue by adrn** [dfm/emcee#289]
tl;dr it looks like `sampler.lnprobability` is transposed in emcee 3 from what I would expect from emcee 2.

Sample code (copy pasta'd from the old docs):
```python
import numpy as np
import emcee

def lnprob(x, ivar):
    return -0.5 * np.sum(ivar * x ** 2)

ndim, nwalkers = 10, 100
ivar = 1. / np.random.rand(ndim)
p0 = [np.random.rand(ndim) for i in range(nwalkers)]

sampler = emcee.EnsembleSampler(nwalkers, ndim, lnprob, args=[ivar])
_ = sampler.run_mcmc(p0, 1000)

print(emcee.__version__)
print(sampler.chain.shape, sampler.lnprobability.shape)
```

## Output from emcee 2.2.1
```
2.2.1
(100, 1000, 10) (100, 1000)
```

## Output from emcee 3.0rc2
```
3.0rc2
(100, 1000, 10) (1000, 100)
```

**Comment by dfm** [comment]
Thanks! This is definitely a bug. The goal is for chain and lnprobability to be backwards compatible so it should have the behaviour you expect!

**Comment by dfm** [comment]
Fixed by #295.
